usr/share/rear/layout/prepare/GNU/Linux/160_include_luks_code.sh

```diff
@@ -34,17 +34,27 @@ create_crypt() {
         esac
     done
 
+    cryptsetup_options+=" $LUKS_CRYPTSETUP_OPTIONS"
+
     (
     echo "Log \"Creating luks device $name on $encdevice\""
     if [ -n "$keyfile" ] ; then
-        echo "cryptsetup luksFormat -q${cipher}${hash}${uuid} ${encdevice} $keyfile"
+        # Assign a temporary keyfile at this stage so that original keyfiles do not leak onto the rescue medium.
+        # The original keyfile will be restored from the backup and then re-assigned to the LUKS device in the
+        # 'finalize' stage.
+        # The scheme for generating a temporary keyfile path must be the same here and in the 'finalize' stage.
+        keyfile="${TMPDIR:-/tmp}/LUKS-keyfile-$(basename $keyfile)"
+        dd bs=512 count=4 if=/dev/urandom of="$keyfile"
```

Comment:
You've stated in _default.conf_:

> but be aware that this produces a
> low-quality master encryption key. For details, see the cryptsetup(8) manual page.

I think that having disks encrypted with key created by /dev/urandom (in case that $target_keyfile does not exist) should be advertised to user as well...

Reply:
Disks are encrypted via a different key: This _master encryption key_ still remains at high quality if `LUKS_CRYPTSETUP_OPTIONS` is used in its default configuration (`--use-random`).

`/dev/urandom` is used here to generate what's basically a very long (2k) temporary password, which remains valid only until the original keyfile will be restored and re-assigned. If this doesn't succeed, the "temporary password" becomes permanent and the following message is issued:
```
LogPrintError "$original_keyfile was not restored - LUKS device $device has been assigned a new keyfile"
```

I would hope that this gives the user a sufficient warning for this improbable case, possibly resulting from a non-functional crypttab configuration anyway. Would you say even more advice is needed?